CMakeLists.txt

```diff
@@ -25,22 +25,39 @@ set( CMAKE_ARCHIVE_OUTPUT_DIRECTORY
    "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}" CACHE PATH
    "Directory for the built static libraries" )
 
-# Set the language standards to use.
-set( CMAKE_CXX_STANDARD 17 CACHE STRING "The (Host) C++ standard to use" )
-set( CMAKE_CUDA_STANDARD 17 CACHE STRING "The (CUDA) C++ standard to use" )
-
-option(DETRAY_ARRAY_PLUGIN "Build std::array math plugin" On)
-option(DETRAY_VECMEM_ARRAY_PLUGIN "Build vecmem::array math plugin" On)
-option(DETRAY_EIGEN_PLUGIN "Build Eigen math plugin" On)
-option(DETRAY_SMATRIX_PLUGIN "Build ROOT/SMatrix math plugin" Off)
-option(DETRAY_VC_ALGEBRA_PLUGIN "Build vc based simd algebra math plugin" ON)
-option(DETRAY_IO_CSV "Build CSV IO module" On)
-option(DETRAY_DISPLAY "Build matplot++ based display module" Off)
-option(DETRAY_BUILD_CUDA "Build the CUDA sources included in detray" Off)
-option(DETRAY_USE_VECMEM "Build with vecmem" On)
+# Include the Detray CMake code.
+list( APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake" )
+include( detray-compiler-options )
+include( detray-functions )
+
+# Check if CUDA is available.
+include( CheckLanguage )
+check_language( CUDA )
+set( DETRAY_BUILD_CUDA_DEFAULT FALSE )
+if( CMAKE_CUDA_COMPILER )
+   set( DETRAY_BUILD_CUDA_DEFAULT TRUE )
+endif()
+
+# Flags controlling which parts of Detray to build.
+option( DETRAY_VECMEM_PLUGIN "Build vecmem::static_array math plugin" ON )
+option( DETRAY_EIGEN_PLUGIN "Build Eigen math plugin" ON )
+option( DETRAY_SMATRIX_PLUGIN "Build ROOT/SMatrix math plugin" OFF )
+option( DETRAY_VC_PLUGIN "Build Vc based math plugin" ON )
+option( DETRAY_IO_CSV "Build CSV IO module" ON )
+option( DETRAY_DISPLAY "Build matplot++ based display module" OFF )
+option( DETRAY_BUILD_CUDA "Build the CUDA sources included in detray"
+   ${DETRAY_BUILD_CUDA_DEFAULT} )
 option( DETRAY_BUILD_TESTING "Build the (unit) tests of Detray"
    TRUE )
-option( DETRAY_BENCHMARKS "Enable benchmark tests" TRUE )
+cmake_dependent_option( DETRAY_BENCHMARKS "Enable benchmark tests" TRUE
+   "DETRAY_BUILD_TESTING" OFF )
+cmake_dependent_option( DETRAY_BENCHMARKS_MULTITHREAD
+   "Enable multithreaded benchmarks" OFF "DETRAY_BENCHMARKS" OFF )
+set( DETRAY_BENCHMARKS_REP 1 CACHE STRING
+   "Number of repetitions for the benchmarks" )
```

Comment:
Nice, I will add the consistency checks here, too, then

Reply:
Consistency checks? :confused:

Reply:
There will be tests that are sort of unittests, but should not be run on the CI. They don't really exist yet (partially in #141 )